packages/launchpad/src/setup-actions.ts

```diff
@@ -0,0 +1,76 @@
+import { isMobile } from '@automattic/viewport';
+import { addQueryArgs } from '@wordpress/url';
+import type { LaunchpadTracksData, Task } from './types';
+
+export const setUpActionsForTasks = (
+	tasks: Task[],
+	siteSlug: string | null,
+	tracksData: LaunchpadTracksData
+): Task[] => {
+	const { recordTracksEvent, checklistSlug, tasklistCompleted } = tracksData;
+
+	//Record click events for tasks
+	const recordTaskClickTracksEvent = ( task: Task ) => {
+		recordTracksEvent( 'calypso_launchpad_task_clicked', {
+			checklist_slug: checklistSlug,
+			checklist_completed: tasklistCompleted,
+			task_id: task.id,
+			is_completed: task.completed,
+			context: 'customer-home',
+		} );
+	};
+
+	// Add actions to known tasks
+	return tasks.map( ( task: Task ) => {
+		let action: () => void;
+
+		//Record task view tracks event
+		recordTracksEvent( 'calypso_launchpad_task_view', {
+			checklist_slug: checklistSlug,
+			task_id: task.id,
+			is_completed: task.completed,
+			context: 'customer-home',
+		} );
```

Comment:
Two things:
1. I think that this Tracks event should be triggered via a `useEffect()` call in the wrapper component so it's correctly (and only) triggered when the component is rendered. It doesn't feel right to be doing this from within this block of code.
2. `context` should come from the caller via `tracksData`.

I also want to note that this implementation approach conflicts with @valterlorran's exploration into adding the task position to the render/view event, as we're doing the sorting after this set of tasks is returned.

Reply:
> I also want to note that this implementation approach conflicts with @valterlorran's exploration into adding the task position to the render/view event, as we're doing the sorting after this set of tasks is returned.

Yeah, I'm going to pause this refactor until https://github.com/Automattic/wp-calypso/pull/79573 ships since it will conflict also.